### DISCUSSION TITLE
采用linux主板的ElectronBot分支

### DISCUSSION THREAD

**Discussion by cgisky** [peng-zhihui/ElectronBot#166] [General]
鉴于对[ElectronBot]的使用场景有一些别的期待，还有就是降低复制难度，并增加一些我想要的功能，决定从这几个方面对其进行魔改。

1. 使用微型linux做为控制板，代替MCU
2. 采用wifi链接，代替USB链接
3. 使用websocket和https协议做为API服务器，替代各语言的SDK。
4. 一个web服务器，用作管理端。同时上位机使用H5编写，用浏览器或electron访问，做到一键跨平台，打通PC各操作系统和手机系统。 
5. 添加语音收放能力，加一个小功放，mic和小腔体音箱
6. 在机器人端 实现一定语音识别、图像识别能力，不必依赖上位机也能实现一定AI能力
7. 身躯不做转动，头部增加转动动作
8. 用SCS0009总线舵机代替稚晖君魔改的舵机
9. 添加电池模块，可以无电源运行。
10. 引出调试接口，方便烧写开发板
11. 增加可编程灯效（双手、身体 添加RGB灯珠）
12. 因为舵机和一些功能增加，需要修改外壳。
13. 考虑外设接口，和可“换装”能力
14. 显示界面采用web界面，增加显示器的玩法，减少编程难度。
15. 显示器换为方形显示器（省钱，显示面积还多点，性价比高）
16. 去除手势感应。
17. 增加外设接口。
18. 一个开放的应用平台
19. ........

-------------------------------------------------以下进度更新--------------------------------------------------------------------

更新日期 2022/4/25
    已完成部分：
     1. 选用 Nanopi Neo Air 做为开发板  linux 4.14 内核，拥有声音、摄像头、wifi 等能力  【√ 完成调试】
     
![1](https://user-images.githubusercontent.com/108901/165070426-bda01c77-1b6d-4556-bbe2-38625506ae5d.png)

     2. 舵机 采用SCS0009总线舵机，tplinker-mini 板为舵机驱动板 【√ 完成调试】
 
![2](https://user-images.githubusercontent.com/108901/165070777-e76cc0c4-b5fa-4001-9caf-1f364dbe6e09.png)

     3. 摄像头采用500W像素的 CAM500B，MJPEG-Stream 做为推流服务器 【√ 完成调试】
 
![3](https://user-images.githubusercontent.com/108901/165070999-33378408-35ec-474c-a1a6-304dd6ff95fd.png)
 
     4. 使用SnowBoy，拥有语音唤醒功能，借鉴 wukong-robot 开发语音能力。【完成度60% 填完驱动的坑，实现语音唤醒，回音消除，后续语音能力待完善】 
     5. 电池管理模块 【√ 完成调试】
     6. 【new】屏幕采用 1.3吋 ST7789VW 240*240 的方形屏幕  SPI链接  【√ 完成调试】
     

https://user-images.githubusercontent.com/108901/165071914-5e183d4f-23bb-4cf8-9a29-d0bbcdffe62f.mp4

  近期待完成部分：
     8. websocket和https协议做为API服务器【○ 未开始】
     9. 管路端、上位机端【○ 未开始】
     12. 外壳修改 【○ 未开始】

近期期待做出第一个原型机来
有对这个方案感兴趣一起搞事情的朋友吗？？

**Comment by coolflyreg** [comment]
用wukong-robot麻烦的是，各个平台过了免费期就不能用了。再申请又很麻烦。有什么本地语音识别和文本转语音的方案吗？

**Reply by cgisky** [reply]
本地机器性能好的话，还是不少开源的语音识别可以跑的。 但是准确率一般在85%左右，和商业的还是有较大差距

**Comment by cgisky** [comment]
CPU全志H3的nanopi neo air，  屏幕显示方面，只能跑QT4。 QT5 或是跑个openbox 小桌面也够呛。
所以换成CPU三星6818 的 Nanopi Core6818，虽然比不上RK3389 但估计应该够用了  成本又贵了100多 o(╥﹏╥)o

**Comment by fcode520** [comment]
我想知道 你这个进度咋样了~~~~持续关注ing

**Reply by GreenShadeZhang** [reply]
> 我想知道 你这个进度咋样了~~~~持续关注ing

我表示也很关注，我都想给电子用树莓派做个轮椅了，然后轮椅作为主控脱离电脑。
